### ISSUE TITLE
Thumbnail scrolling speed under linux too fine.

### ISSUE THREAD

**Issue by abertens** [Beep6581/RawTherapee#4663]
Hi there!

Today I git clone the dev branch to try the latest functionality and I found an enhancement in the gui that not works well on my linux desktop.

Merge pull request #4652 from Thanatomanic/scroll-speed (Scrolling speed changed to height of thumbnail) 8a9f102cd252c26629dd2bc314e8b3ff9230e0ca, from Jul 1, 2018 makes thumbnail scrolling almost unusable on a directory with 200 images. The slightest touch on the scroll button moves the page upward/downward that makes me stop using it and start using keys instead to have some control over it.

May I ask if you tested this merge on linux? I don't know about Windows...

Information about my build:
Version: 5.4-470-gb419bb2d
Branch: dev
Commit: b419bb2d
Commit date: 2018-07-04
Compiler: cc 5.5.0
Processor: x86_64
System: Linux
Bit depth: 64 bits
Gtkmm: V3.22.2
Lensfun: V0.3.2.0
Build type: Release
Build flags:  -std=c++11 -march=native -Werror=unused-label -flto -fopenmp -Werror=unknown-pragmas -Wall -Wno-unused-result -Wno-deprecated-declarations -O3 -DNDEBUG -ftree-vectorize
Link flags:  -march=native -flto
OpenMP support: ON
MMAP support: ON

Gtk version is gtk+-3.22.30.

Thank you for your effort to bring us this magnificent tool.

Yours,
Andres Bertens

**Comment by abertens** [comment]
Additional info:
Testing on a directory with a 25 images behaves the same. Seems that image number is irrelevant.

Forgot to mention that reverting merge 8a9f102cd252c26629dd2bc314e8b3ff9230e0ca makes it usable again.

Thanks.

**Comment by Thanatomanic** [comment]
Hi! Thanks for reporting. Can you confirm if scrolling in your 200 image folder still goes in discrete steps? I mean, it should scroll now in steps of the thumbnail height. Does it do that?

**Comment by ff2000** [comment]
"The slightest touch on the scroll button moves the page upward/downward"
Are you talking of the scroll wheel of your mouse? Or the slider on the scrollbar?
I am also on gtk+-3.22.30 on Gentoo linux and have no issues. I only noticed a small inconsistency: Turning the scroll wheel while mouse is over the thumb view each turn scrolls by one row. But turning while over the scrollbar reveals the old "micro-scrolling".

[As a small request: scrolling by max. visible rows would be awesome, too. If 3.5 rows are visible (4th row is cut off) scroll by 3 rows so that the 4th is now on top. Would make quick brwosing even quicker ;)]

**Comment by abertens** [comment]
Sorry to not be clear enough in my explanation, it is hard to explain something so natural...
Perhaps is not a problem at all and is the functionality you were looking for.

It is not a problem when you scroll with mouse using the scroll bars at window right side.
Using mouse scroll wheel is my problem, getting much worse when using the scroll wheel emulation that a touchpad offers.

Yes, you are correct. The scrolling using mouse scroll wheel is discrete now: one row of thumbnails replaces the next (or previous one). Looks like now there is no intermediate positions.

Using the scroll wheel I can not easily follow images as one row is replaced by the next one.
I have to re-view and re-think because image disappears from actual row and appears on next one when I need fine control.

Thanks for your hint: indeed turning while over the scrollbar reveals the old "micro-scrolling". But I have to move to the scrollbar to use it and does not look/seems natural this additional move.

Perhaps is a matter of tastes and get used to it but I certainly prefer the old behaviour. 

Thanks.

**Comment by ff2000** [comment]
No problem.
Yes, it needs getting used to it. I also like smooth scrolling more as you can follow more easily the flow of images (the old behaviour also wasn't really smooth btw). No idea if it's just a property that needs to be enabled or if it has to be done by hand - I have no real idea concerning GTK+.

**Comment by Thanatomanic** [comment]
@abertens @ff2000 I agree that smooth scrolling would be the more natural behavior when using a touchpad or touchscreen for scrolling, instead of a mouse. However, I think it's highly unlikely that different behaviors can be implemented based on the hardware used to scroll.
So I think it has to come down to a matter of taste, or rethinking the smooth scrolling again. Because, as ff2000 said, it wasn't really smooth to begin with.

Could anyone from the main devs make a decision here?

**Comment by Beep6581** [comment]
The current behavior:
* Vertical scrolling:
  * Thumbs scroll by exactly the row height when using the mouse scroll-wheel over the thumbs. Since one row gets perfectly replaced by the next, this looks nice and seems that everyone likes it except for @abertens 
  * Thumbs scroll by about 45% of their row height when using the mouse scroll-wheel over the vertical scrollbar. This is neither smooth nor consistent.
* Horizontal scrolling:
  * Thumbs scroll by some amount, perhaps the row height, when using the mouse scroll-wheel over the thumbs. It would be difficult to have consistent behavior with vertical scrolling since a thumb's height is constant while the width is not: 
![screenshot_20180821_132150](https://user-images.githubusercontent.com/5844619/44398717-2e205f00-a545-11e8-92fc-ec0efa922488.png)
  * When using the mouse scroll-wheel over the horizontal scrollbar, thumbs scroll by about 98% of what they would have scrolled if you had hovered the mouse over the thumbs instead.

There are two questions here:
1. Do we want smooth-scrolling or jumping?
2. How much should the total area scrolled by one movement of the wheel be? i.e. even with smooth-scrolling it could still scroll one thumb row height at a time.

I am fine with jumping if it jumps one thumb row height at a time, but I would prefer smooth-scrolling if it moved any other amount. Currently vertical scrolling jumps one row height at a time, so it's good.

I tested mouse scroll-wheel scrolling vs laptop touchpad scrolling - both worked identically. Sabayon/KDE.

The remaining issue is that scrolling over the thumbs scrolls a different amount than scrolling over the scrollbars. This could be due to a GTK+ bug, or it could be that the two are/need to be handled differently, quote from GTK+ devs in IRC:
```
 baedert | afaik that's just a bug that nobody has bothered to fix
~Company | probably because the scrollbar has its own scrolling handler?
~Company | no, I'm talking about GtkScrollbar having its own code for scrolling
~Company | which overrides the one in GtkScrolledWindow
~Company | easiest way to check this would be to try the scrollwheel on the horizontal scrollbar in the bottom I guess
```
